File: app/src/main/java/com/hidesign/ported/adapters/RecyclerAdapter.java
package com.hidesign.ported.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class RecyclerAdapter extends RecyclerView.Adapter<RecyclerAdapter.ViewHolder> {

    private OnRecyclerAdapterListener mCallback;
    public interface OnRecyclerAdapterListener {
        void onRecyclerAdapterCreated(RecyclerAdapter adapter, ViewHolder v, int position);
    }

    public void setOnRecyclerAdapterListener(OnRecyclerAdapterListener listener) {

        try {
            mCallback = listener;
        } catch (ClassCastException e)  {
            throw new ClassCastException(this.toString() + " must implement OnRecyclerAdapterListener");
        }
    }

    private int resId;
    private int count;

    public class ViewHolder extends RecyclerView.ViewHolder {

        public View view;
        ViewHolder(View v) {
            super(v);
            view = v;
        }
    }

    public RecyclerAdapter(int count, int resId) {
        this.count = count;
        this.resId = resId;
    }

    @NonNull
    @Override
    public RecyclerAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(resId, parent, false);
        return new ViewHolder(v);
    }

    // Replace the contents of a view (invoked by the layout manager)
    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        // - get element from your dataset at this position
        // - replace the contents of the view with that element
//        holder.mTextView.setText(mDataset[position]);
        if(mCallback != null) {
            mCallback.onRecyclerAdapterCreated(this, holder, position);
        }
    }

    @Override
    public int getItemCount() {
        return count;
    }
}

File: app/src/main/java/com/hidesign/ported/adapters/RecyclerAdapter.kt
package com.hidesign.ported.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView

class RecyclerAdapter(private val count: Int, private val resId: Int) : RecyclerView.Adapter<RecyclerAdapter.ViewHolder>() {

    private var mCallback: OnRecyclerAdapterListener? = null

    interface OnRecyclerAdapterListener {
        fun onRecyclerAdapterCreated(adapter: RecyclerAdapter?, v: ViewHolder?, position: Int)
    }

    fun setOnRecyclerAdapterListener(listener: OnRecyclerAdapterListener?) {
        mCallback = try {
            listener
        } catch (e: ClassCastException) {
            throw ClassCastException("$this must implement OnRecyclerAdapterListener")
        }
    }

    inner class ViewHolder internal constructor(var view: View) : RecyclerView.ViewHolder(view)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val v = LayoutInflater.from(parent.context).inflate(resId, parent, false)
        return ViewHolder(v)
    }

    // Replace the contents of a view (invoked by the layout manager)
    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        // - get element from your dataset at this position
        // - replace the contents of the view with that element
//        holder.mTextView.setText(mDataset[position]);
        if (mCallback != null) {
            mCallback!!.onRecyclerAdapterCreated(this, holder, position)
        }
    }

    override fun getItemCount(): Int {
        return count
    }
}